oak_functions/abi/src/lib.rs

```diff
@@ -48,6 +49,15 @@ pub mod proto {
     }
 }
 
+/// Holds the `raw_label` and the `value` to report a metric.
+#[derive(Serialize, Deserialize)]
+pub struct ReportMetricRequest {
```

Comment:
I think this should be feature-gated. It is weird to have this struct here in the cases where the metrics features is not enabled.

Reply:
It seems that if I feature-gate `ReportMetricRequest` here, then I have in `metric.rs` an "unresolved import `oak_functions_abi::ReportMetricRequest`". 

Feature-gating in `metric.rs` seems wrong to me.

Am I missing something here?

Reply:
The entire metric.rs is already feature-gated in https://github.com/project-oak/oak/blob/main/oak_functions/loader/src/lib.rs#L25, so it should not try to build it unless the feature is enabled.

I think you need to define a feature called `oak-metrics` in the config.toml of the abi, and also update the features in the loader and SDK config.toml files to enable the oak-metrics abi feature if the oak-metrics feature is enabled on the loader, something like:
`oak-metrics = ["oak_functions_abi/oak-metrics"]`

Reply:
Personally I wouldn't bother feature gating a plain struct (which will be optimized away anyways).